sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala

```diff
@@ -135,11 +135,25 @@ private[sql] object InferSchema {
           // when we see a Java BigInteger, we use DecimalType.
           case BIG_INTEGER | BIG_DECIMAL =>
             val v = parser.getDecimalValue
-            DecimalType(v.precision(), v.scale())
+            try {
+              // Creating `DecimalType` here can fail when precision is bigger than 38.
+              DecimalType(v.precision(), v.scale())
```

Comment:
It's reasonable to have a decimal "0.01", so we should use max of precision and scale as precision here. Also it's better  to not reply on exception for branch.

if (Math.max(v.precision(), v.scale()) <= DecimalType.MAX_MAX_PRECISION) {
  DecimalType(Math.max(v.precision(), v.scale()), v.scale())
} else {
  DoubleType
}